### PR TITLE
DOCSP-43069: Replace documents

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-43069
Staging - https://deploy-preview-52--docs-c.netlify.app/write/replace/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/write/replace.txt [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+   :values: reference
+
+.. meta::
+   :keywords: modify, change, code example
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-short+} to run a replace
+operation on a MongoDB collection. A replace operation performs differently
```
Comment (rustagir) [inline source/write/replace.txt:24]:
```suggestion
In this guide, you can learn how to use the {+driver-short+} to run a **replace
operation** on a MongoDB collection. A replace operation performs differently
```

File: source/write/replace.txt [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+This function removes all fields except the ``_id`` field from the first document that
+matches the search criteria. It then inserts the fields and values you specify into the 
+document.
+
+Required Parameters
+~~~~~~~~~~~~~~~~~~~
+
+The ``mongoc_collection_replace_one()`` function requires the following parameters:
+
+- Collection: Specifies the collection to update.
```
Comment (rustagir) [inline source/write/replace.txt:52]:
```suggestion
- Collection: Specifies which collection to perform the replace operation on.
```

File: source/write/replace.txt [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+- Query filter document: Specifies which collection documents to update. For
+  more information about query filters, see the 
+  :manual:`Query Filter Documents section </core/document/#query-filter-documents>` in
+  the {+mdb-server+} manual.
```
Comment (rustagir) [inline source/write/replace.txt:57]:
```suggestion
- Query filter document: Specifies which collection documents to match. The function selects the first
  matching document to replace. For
  more information about query filters, see the 
  :manual:`Query Filter Documents section </core/document/#query-filter-documents>` in
  the {+mdb-server+} manual.
```

File: source/write/replace.txt [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+- Results location: Specifies a pointer to overwritable storage that will contain
+  operation results, or ``NULL``.
+
+- Error location: Specifies a location for an error value, or ``NULL``.
+
+Example
+~~~~~~~
+
+The following example uses the ``mongoc_collection_replace_one()`` function to replace the fields and values
+of a document in which the ``name`` field value is ``"Pizza Town"``:
```
Comment (rustagir) [inline source/write/replace.txt:71]:
```suggestion
The following example uses the ``mongoc_collection_replace_one()`` function to replace the fields and values
of a document in which the ``name`` field value is ``"Pizza Town"`` with a document in which the name is ...:
```

File: source/write/replace.txt [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+.. literalinclude:: /includes/write/replace.c
+    :language: c
+    :dedent:
+    :start-after: start-replace-one
+    :end-before: end-replace-one
+
+.. important::
+
+   The values of ``_id`` fields are immutable. If your replacement document specifies 
+   a value for the ``_id`` field, it must match the ``_id`` value of the existing document.
```
Comment (rustagir) [inline source/write/replace.txt:82]:
```suggestion
   a value for the ``_id`` field, it must be identical to the ``_id`` value of the existing document.
```

File: source/write/replace.txt [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+         For more information, see the :manual:`hint statement </reference/command/update/#std-label-update-command-hint>`
+         in the {+mdb-server+} manual.
+
+   * - ``comment``
+     - | Attaches a comment to the operation. For more information, see the :manual:`insert command
+         fields </reference/command/insert/#command-fields>` guide in the
+         {+mdb-server+} manual.
+
+Example
+~~~~~~~
```
Comment (rustagir) [inline source/write/replace.txt:129]:
```suggestion
Replace Options Example
~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by rustagir — CHANGES_REQUESTED** [review]
a few small things

**Review by rustagir — APPROVED** [review]

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**